server/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -339,7 +338,7 @@ public void addIndexOperationListener(IndexingOperationListener listener) {
      * @param name Name of the SimilarityProvider
      * @param similarity SimilarityProvider to register
      */
-    public void addSimilarity(String name, TriFunction<Settings, Version, ScriptService, Similarity> similarity) {
+    public void addSimilarity(String name, TriFunction<Settings, IndexVersion, ScriptService, Similarity> similarity) {
```

Comment:
This looks to be a plugin method, but we don't guarantee compatibility with plugins between versions. I don't know if this is likely to cause significant issues though, if we need to provide a backwards-compatible method for `Version`.